### PR TITLE
Allow groups to be pre-collapsed in JSON

### PR DESCRIPTION
**Pull request by jeffleeismyhero**
By adding `"collapsed": true` to a group's JSON, you can specify which groups should be collapsed by default.

**Example JSON**

```json
[
  {
    "id":"personal-details",
    "title":"Personal Details",
    "collapsed":true,
    "sections":[
      {
        "id":"personal-details-0",
        "fields":[
          {
            "id":"first_name",
            "title":"First name",
            "info":"Fornavn",
            "type":"name",
            "size":{
              "width":30,
              "height":1
            },
            "validations":{
              "required":true,
              "min_length":2
            },
            "targets":[
              {
                "id":"display_name",
                "type":"field",
                "action":"update",
                "condition":"$username == 0"
              }
            ]
          },
          {
            "id":"last_name",
            "title":"Last name",
            "type":"name",
            "info":"Obs! Husk å kjøpe melk",
            "size":{
              "width":30,
              "height":1
            },
            "validations":{
              "required":true,
              "min_length":2
            },
            "targets":[
              {
                "id":"display_name",
                "type":"field",
                "action":"update",
                "condition":"$username == 0"
              }
            ]
          },
          {
            "id":"display_name",
            "title":"Display name",
            "info":"This is the name that is going to be used across the application.",
            "type":"text",
            "size":{
              "width":40,
              "height":1
            },
            "formula":"first_name last_name"
          }
        ]
      },
      {
        "id":"personal-details-1",
        "fields":[
          {
            "id":"address",
            "title":"Address",
            "type":"text",
            "size":{
              "width":50,
              "height":1
            }
          },
          {
            "id":"email",
            "title":"E-post",
            "info":"bork\nbork\nbork",
            "type":"email",
            "size":{
              "width":25,
              "height":1
            },
            "validations":{
              "format":"[\\w._%+-]+@[\\w.-]+\\.\\w{2,}",
              "required":true
            }
          },
          {
            "id":"username",
            "title":"Username",
            "type":"select",
            "size":{
              "width":25,
              "height":1
            },
            "values":[
              {
                "id":0,
                "title":"Generated",
                "info":"Generate using firstname & lastname",
                "default":true,
                "targets":[
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"enable"
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"enable"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"disable"
                  }
                ]
              },
              {
                "id":1,
                "title":"Override",
                "info":"Enable manual editing for the display name",
                "targets":[
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"enable"
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"enable"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"enable"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"update"
                  }
                ]
              },
              {
                "id":2,
                "title":"Mr.Melk",
                "info":"Bork!",
                "targets":[
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"disable"
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"disable"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"disable"
                  },
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"update",
                    "target_value":"Mr."
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"update",
                    "target_value":"Melk"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"update"
                  },
                  {
                    "id":"contract_type",
                    "type":"field",
                    "action":"update",
                    "target_value":0
                  }
                ]
              },
              {
                "id":3,
                "title":"John Hyperseed",
                "info":"Other info",
                "targets":[
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"disable"
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"disable"
                  },
                  {
                    "id":"first_name",
                    "type":"field",
                    "action":"update",
                    "target_value":"John"
                  },
                  {
                    "id":"last_name",
                    "type":"field",
                    "action":"update",
                    "target_value":"Hyperseed"
                  },
                  {
                    "id":"display_name",
                    "type":"field",
                    "action":"update"
                  },
                  {
                    "id":"contract_type",
                    "type":"field",
                    "action":"update",
                    "target_value":1
                  }
                ]
              }
            ]
          },
          {
            "id":"image",
            "type":"image",
            "size":{
              "width":100,
              "height":1.7
            }
          }
        ]
      }
    ]
  },
  {
    "id":"employment",
    "title":"Employment",
    "sections":[
      {
        "id":"employment-0",
        "fields":[
          {
            "id":"start_date",
            "title":"Start date",
            "info":"The day of employment",
            "type":"date",
            "size":{
              "width":25,
              "height":1
            }
          },
          {
            "id":"start_time",
            "title":"Start time",
            "type":"time",
            "size":{
              "width":25,
              "height":1
            }
          },
          {
            "id":"end_date",
            "title":"End date",
            "type":"date",
            "validations":{
              "compare_to":"start_date",
              "compare_rule":">"
            },
            "size":{
              "width":25,
              "height":1
            }
          },
          {
            "id":"end_time",
            "title":"End time",
            "type":"time",
            "size":{
              "width":25,
              "height":1
            },
            "validations":{
              "required":true,
              "min_length":2
            }
          },
          {
            "id":"contract_type",
            "title":"Contract type",
            "info":"Type of contract\n Permanent or temporary",
            "type":"select",
            "size":{
              "width":50,
              "height":1
            },
            "values":[
              {
                "id":0,
                "title":"Permanent",
                "info":"Regular employee",
                "default":true,
                "targets":[
                  {
                    "id":"end_date",
                    "type":"field",
                    "action":"show"
                  },
                  {
                    "id":"end_time",
                    "type":"field",
                    "action":"show"
                  },
                  {
                    "id":"employment-1",
                    "type":"section",
                    "action":"show"
                  }
                ]
              },
              {
                "id":1,
                "title":"Temporary",
                "targets":[
                  {
                    "id":"end_date",
                    "type":"field",
                    "action":"hide"
                  },
                  {
                    "id":"end_time",
                    "type":"field",
                    "action":"hide"
                  },
                  {
                    "id":"employment-1",
                    "type":"section",
                    "action":"hide"
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        "id":"employment-1",
        "fields":[
          {
            "id":"base_salary",
            "title":"Base salary",
            "type":"select",
            "size":{
              "width":25,
              "height":1
            },
            "values":[
              {
                "title":"Salary 1",
                "id":0,
                "value":100,
                "targets":[
                  {
                    "id":"total",
                    "type":"field",
                    "action":"update"
                  }
                ]
              },
              {
                "title":"Salary 2",
                "id":1,
                "value":200,
                "targets":[
                  {
                    "id":"total",
                    "type":"field",
                    "action":"update"
                  }
                ]
              },
              {
                "title":"Salary 3",
                "id":2,
                "value":10,
                "targets":[
                  {
                    "id":"total",
                    "type":"field",
                    "action":"update"
                  }
                ]
              }
            ]
          },
          {
            "id":"bonus_enabled",
            "title":"Bonus enabled",
            "type":"select",
            "size":{
              "width":25,
              "height":1
            },
            "values":[
              {
                "id":true,
                "title":"Ja"
              },
              {
                "id":false,
                "title":"Nei"
              }
            ]
          },
          {
            "id":"bonus",
            "title":"Bonus",
            "type":"float",
            "size":{
              "width":25,
              "height":1
            },
            "targets":[
              {
                "id":"total",
                "type":"field",
                "action":"update"
              }
            ]
          },
          {
            "id":"total",
            "title":"Total",
            "type":"float",
            "size":{
              "width":25,
              "height":1
            },
            "disabled":true,
            "formula":"$base_salary + $bonus"
          }
        ]
      }
    ]
  },
  {
    "id":"companies",
    "title":"Companies",
    "sections":[
      {
        "id":"companies[0]",
        "fields":[
          {
            "id":"companies[0].name",
            "title":"Name 1",
            "info":"Hyper",
            "type":"name",
            "validations":{
              "required":true,
              "min_length":2
            },
            "size":{
              "width":50,
              "height":1
            }
          },
          {
            "id":"companies[0].phone_number",
            "title":"Phone number 1",
            "type":"number",
            "size":{
              "width":30,
              "height":1
            }
          },
          {
            "id":"companies[0].remove",
            "title":"Remove",
            "type":"button",
            "size":{
              "width":20,
              "height":1
            }
          }
        ]
      },
      {
        "id":"companies[1]",
        "fields":[
          {
            "id":"companies[1].name",
            "title":"Name 2",
            "type":"name",
            "validations":{
              "required":true,
              "min_length":2
            },
            "size":{
              "width":50,
              "height":1
            }
          },
          {
            "id":"companies[1].phone_number",
            "title":"Phone number 1",
            "type":"number",
            "size":{
              "width":30,
              "height":1
            }
          },
          {
            "id":"companies[1].remove",
            "title":"Remove",
            "type":"button",
            "size":{
              "width":20,
              "height":1
            }
          }
        ]
      }
    ]
  }
]
```

### CODE REVIEW

**Comment by 3lvis** [comment]
@jeffleeismyhero Wouldn't it be better to have valid groups collapsed by default? Do you have any real-use cases for this?

Looks good though :)

**Comment by jeffleeismyhero** [comment]
I would imagine that one would want to collapse groups that are optional and/or rarely used. In my case, this would be "Adjustments" for an order. By and large, most groups need to be expanded and visible so the default should be expanded -- this also makes it a non-breaking change.

ios simulator screen shot jul 28 2015 7 09 19 pm

File: Source/Models/FORMGroup.m [hunk 1]
```diff
@@ -54,6 +56,10 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
     return self;
 }
 
+- (void)updateGroupWithCollapsed:(BOOL)collapsed {
+    abort();
```
Comment (3lvis) [inline Source/Models/FORMGroup.m]: What's this for?

File: Source/Models/FORMGroup.m [hunk 2]
```diff
@@ -54,6 +56,10 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
     return self;
 }
 
+- (void)updateGroupWithCollapsed:(BOOL)collapsed {
```
Comment (zenangst) [inline Source/Models/FORMGroup.m]: @jeffleeismyhero Looks like you could remove this method declaration, I can't find any references to it.

**Comment by jeffleeismyhero** [comment]
It seems like these tests are failing because of a CI issue, not a code issue.

**Comment by 3lvis** [comment]
WTF Circle CI?

**Comment by jeffleeismyhero** [comment]
Any idea if this is CI related or something I would need to fix?

**Comment by zenangst** [comment]
@jeffleeismyhero hey, I'm gonna take a look, I'll keep you posted :sunglasses:

**Comment by jeffleeismyhero** [comment]
Bump :)

**Comment by 3lvis** [comment]
Stupid Circle CI

**Comment by zenangst** [comment]
I have disabled Circle CI now.

**Pull request merged**